### ISSUE TITLE
Unit tests fail with 'UnhandledPromiseRejectionWarning' when tsconfig module type is 'es2015'

### ISSUE THREAD

**Issue by pmarflee** [alsatian-test/alsatian#431]
I am trying to use Alsatian to unit test my [solutions](https://github.com/pmarflee/AdventOfCode2017-TypeScript) to this year's Advent of Code problems.  However, when I try to run my tests, I receive the following error:

> (node:8316) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): SyntaxError: Unexpected token import
(node:8316) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

I was able to get my tests to run by changing the module type in tsconfig.json from 'es2015' to 'commonjs'.  However this broke one of the Vue.js components that my UI depends on, so I'd prefer to keep these settings as they are.

I'm not an expert in TypeScript, so any help would be appreciated in finding a solution.

**Comment by Jameskmonger** [comment]
What command are you using to run the tests?

**Comment by pmarflee** [comment]
@Jameskmonger  Here's a link to my [package.json](https://github.com/pmarflee/AdventOfCode2017-TypeScript/blob/master/package.json)

**Comment by seiyria** [comment]
I'm also seeing this problem. Here is my tsconfig:

```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "src",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "module": "commonjs",
    "rootDir": "src/",
    "outDir": "./dist",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "ES2015",
    "skipLibCheck": true,
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2016",
      "dom"
    ]
  },
  "exclude": [
    "node_modules",
    "dist",
    "client"
  ]
}
```

I'm trying to run `alsatian \"./test/**/*.spec.ts\"` and I have one sample test:

```

import { TestFixture, TestCase, Test, Expect } from 'alsatian';

@TestFixture('Spell effect verification')
export class EffectVerifier {

  @TestCase(2, 3)
  @Test('test')
  public test(a, b) {
    Expect(a).toBe(b);
  }
  
}
```

Any ideas why it's not working?

**Comment by Jameskmonger** [comment]
I _think_ that this is caused because you're compiling to ES2015 but not using a version of Node that supports ES2015 syntax.

@jamesrichford Haven't we run into this issue before?

**Comment by seiyria** [comment]
I'm on 8.9.3 right now, for what it's worth. I think one of my dependencies is blocking an upgrade to 9. So, if that's the case, any workarounds would be appreciated too.

**Comment by jamesrichford** [comment]
@seiyria @pmarflee we're working on a PR to give more information about these errors but incidentally I think your problem might be solved by going with the tsnode approach as this runs the test on typescript rather than needing the step to compile them to javascript. An example is here https://github.com/alsatian-test/alsatian/issues/419 :)

**Comment by Jameskmonger** [comment]
This should be solved now in `2.1.0` - errors are now logged properly to the console.

@pmarflee @seiyria Can you use `alsatian@2.1.0` and let me know?

**Comment by jamesadarich** [comment]
Closing due to inactivity. Should be fixed. Please reopen if not
